chainntnfs/txnotifier.go

```diff
@@ -763,9 +763,7 @@ func (n *TxNotifier) CancelConf(confRequest ConfRequest, confID uint64) {
 	if confSet.details != nil {
 		confHeight := confSet.details.BlockHeight +
 			ntfn.NumConfirmations - 1
-		if confHeight <= n.currentHeight {
```

Comment:
@wpaulino Just to make sure: was this condition here for a good reason?

Reply:
not 100% sure myself, but i think your change is correct. doesn't make sense to continue tracking a canceled request

Reply:
It looks like I actually meant to write the opposite condition. If the confirmation height has not been reached, then we should delete it as it hasn't been notified yet to prevent the notification, but if it has, then there's no need to delete it as that's already done once the notification is dispatched. Either way, it should be fine to just always delete now.